lib/evaluate/check-expression.cc

```diff
@@ -237,4 +261,93 @@ template void CheckSpecificationExpr(const std::optional<Expr<SomeInteger>> &,
 template void CheckSpecificationExpr(
     const std::optional<Expr<SubscriptInteger>> &, parser::ContextualMessages &,
     const semantics::Scope &);
+
+// IsSimplyContiguous() -- 9.5.4
+class IsSimplyContiguousHelper
+  : public AnyTraverse<IsSimplyContiguousHelper, std::optional<bool>> {
+public:
+  using Result = std::optional<bool>;  // tri-state
```

Comment:
Why can't `Result` be just `bool`? When is does it have the value`std::nullopt`?

Reply:
It's a tri-state value: false, true, or unknown.